### ISSUE TITLE
Support mongoose 5

### ISSUE THREAD

**Issue by gforge** [dsanel/mongoose-delete#36]
Just noticed that mongoose 5 was released. Here's the how-to-upgrade: https://github.com/Automattic/mongoose/blob/master/migrating_to_5.md

After a quick glance I don't think there should be any issues with upgrading :-)

**Comment by gforge** [comment]
Since the problem with 5.0.7 and above is a bug I figure we continue the discussion in the issue section instead of the PR.

I've created pure test case for `mongoose` (planned adding it to an `mongoose` issue, *note* change `mongoose.model` to `db.model` when testing in the mongoose-package):

```
it('statics', async () => {
  var called = false;
  var test_plugin = function(schema) {
    schema.statics.count = function() {
      called = true;
      return mongoose.Model['count'].apply(this, arguments).where('filter').eq(true);
    }
  }

  var TestSchema = new Schema({ name: String, filter: Boolean });
  TestSchema.plugin(test_plugin);
  var TestModel = mongoose.model('PluginTests', TestSchema);

  await TestModel.create(
      [
          { name: 'Obi-Wan Kenobi', filter: true},
          { name: 'Darth Vader', filter: false},
          { name: 'Luke Skywalker', filter: true}
      ]);

  assert.equal(called, false);
  var count = await TestModel.count();
  assert.equal(called, true);
  assert.equal(count, 2);
});
```

Annoyingly this seems to pass without any issues.

**Comment by gforge** [comment]
Ok, so I have found the culprit: passing a function fails due to the `kareem` update in version 2.0.5:  https://github.com/vkarpov15/kareem/commit/5afc5b9ab34f82702c8cd3a700c0f67fdfe4c86d#diff-168726dbe96b3ce427e7fedce31bb0bcR307

Thus when I use the following test case it fails:

```
it('statics', async (done) => {
  var called = false;
  var test_plugin = function(schema) {
    schema.statics.count = function() {
      called = true;
      return mongoose.Model['count'].apply(this, arguments).where('filtered').ne(true);
    }
  }

  var TestSchema = new Schema({ name: String, filtered: Boolean }, { collection: 'mongoose_delete_test10'});
  TestSchema.plugin(test_plugin);
  var TestModel = mongoose.model('PluginTests', TestSchema);

  await TestModel.create(
      [
          { name: 'Obi-Wan Kenobi', filtered: true},
          { name: 'Darth Vader'},
          { name: 'Luke Skywalker', filtered: true}
      ]);

  assert.equal(called, false);
  return TestModel.count((err, count) => {
    assert.equal(called, true);
    assert.equal(count, 1);
    done();
  });
});
```

**Comment by vkarpov15** [comment]
This issue will be fixed with mongoose 5.0.12. Thanks for reporting :+1:

**Comment by nishant-lbb** [comment]
Is the support available now for mongoose 5+ ?

**Comment by vkarpov15** [comment]
@nishant-lbb should be as long as you're using `mongoose@^5.0.12`

**Comment by dsanel** [comment]
Fixed in `v0.5.0`
